app/static/js/userProfile.js

```diff
@@ -1,323 +1,367 @@
+          console.log(error)
+          msgToast("Error!", "Failed to save changes!")
+        }
+    });
+  })
+
+  $("#printButton").on("click", function() {
+        let username = $(this).data('username')
+        printDocument(`/profile/${username}/travelForm`)
+      })
+  $("#actions").on("change", changeAction)
+  $("#phoneInput").inputmask('(999)-999-9999');
+  $(".notifyInput").click(function updateInterest(){
+    var programID = $(this).data("programid");
+    var username = $(this).data('username');
+
+
+    var interest = $(this).is(':checked');
+    var routeUrl = interest ? "addInterest" : "removeInterest";
+    interestUrl = "/" + username + "/" + routeUrl + "/" + programID ;
+    $.ajax({
+      method: "POST",
+      url: interestUrl,
+      success: function(response) {
+          reloadWithAccordion("programTable")  //  Reloading page after user clicks on the show interest checkbox
+      },
+      error: function(request, status, error) {
+        console.log(status,error);
+        location.reload();
+      }
+    });
+  });
+
+  $('.removeFromTranscriptCheckbox').click(function() {
+    var removeFromTranscript = $(this).is(':checked');
+    var username = $(this).data('username');
+
+
+    $.ajax({
+        type: "POST",
+        url: `/profile/${username}/updateTranscript/${programID}`,  
+        contentType: "application/json",
+        data: JSON.stringify({ username: username, removeFromTranscript: removeFromTranscript, programID: programID }),
+        success: function(response) {
+            console.log(response);
+            // Handle success response if needed
+        },
+        error: function(error) {
+            console.error("An error occurred:", error);
+            // Handle error if needed
+        }
+    });
+  });
+
+
+  function changeAction(e){
+    let profileAction = $(this).val()
+    let username = $(this).data('username')
+    if (profileAction == "Emergency Contact"){
+      window.location.href = `/profile/${username}/emergencyContact`
+    } else if (profileAction == "Insurance Information"){
+      window.location.href = `/profile/${username}/insuranceInfo`
+    } else if(profileAction == "Print Travel Form"){
+      printDocument(`/profile/${username}/travelForm`)
+    } else if (profileAction == "View Service Transcript"){
+      window.location.href = `/profile/${username}/serviceTranscript`
+    } else if (profileAction == "Manage CCE Minor") {
+      window.location.href = `/profile/${username}/cceMinor`
+    }
+    $(this).val('')
+  }
+
+  // This function is to disable all the dates before current date in the ban modal End Date picker
+  $(function(){
+    var banEndDatepicker = $("#banEndDatepicker");
+    banEndDatepicker.datepicker({
+      changeYear: true,
+      changeMonth: true,
+      minDate:+1,
+      dateFormat: "yy-mm-dd",
+    }).attr('readonly','readonly');
+  });
+
+    /*
+     * Ban Functionality
+     */
+
+
+  var programID;
+  $(".ban").click(function() {

```

Comment:
I'm pretty sure this wasn't you all but I think you should be the changemakers to pave our bright future and change the name of this class from `ban` to `banEdit` or something to that effect. The class really doesn't seem like it should be applied to the edit button. See line 205 in `userProfile.html` for the code that sets this class. I think its only referenced here from what I could tell so it should be a quick change.

Reply:
Fixed